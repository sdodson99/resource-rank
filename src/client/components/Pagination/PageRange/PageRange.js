import React from 'react';
import Page from '../Page/Page';
import PropTypes from 'prop-types';

const PageRange = ({ first, last, onClick }) => {
  const pages = [];

  for (let page = first; page <= last; page++) {
    pages.push(<Page key={page} number={page} onClick={onClick} />);
  }

  return pages;
};

PageRange.propTypes = {
  first: PropTypes.number,
  last: PropTypes.number,
  onClick: PropTypes.func,
};

PageRange.defaultProps = {
  first: 1,
  last: 2,
};

export default PageRange;
