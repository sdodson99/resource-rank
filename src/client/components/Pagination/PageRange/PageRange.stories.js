import React from 'react';
import PageRange from './PageRange';

export default {
  component: PageRange,
  title: 'Components/Pagination/PageRange',
  decorators: [
    (Story) => {
      return (
        <div className="flex">
          <Story />
        </div>
      );
    },
  ],
};

const Template = (args) => <PageRange {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Many = Template.bind({});
Many.args = {
  last: 20,
};
