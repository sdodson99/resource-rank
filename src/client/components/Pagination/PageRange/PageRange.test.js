import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import PageRange from './PageRange';

describe('<PageRange />', () => {
  it('should mount', () => {
    render(<PageRange />);

    const pageRange = screen.getAllByTestId('Page');

    expect(pageRange.length).toBe(2);
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<PageRange />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with many pages', () => {
    const component = createRenderer().render(<PageRange last={20} />);

    expect(component).toMatchSnapshot();
  });
});
