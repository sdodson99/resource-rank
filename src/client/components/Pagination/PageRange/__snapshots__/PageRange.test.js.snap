// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageRange /> should render correctly 1`] = `
Array [
  <Page
    number={1}
  />,
  <Page
    number={2}
  />,
]
`;

exports[`<PageRange /> should render correctly with many pages 1`] = `
Array [
  <Page
    number={1}
  />,
  <Page
    number={2}
  />,
  <Page
    number={3}
  />,
  <Page
    number={4}
  />,
  <Page
    number={5}
  />,
  <Page
    number={6}
  />,
  <Page
    number={7}
  />,
  <Page
    number={8}
  />,
  <Page
    number={9}
  />,
  <Page
    number={10}
  />,
  <Page
    number={11}
  />,
  <Page
    number={12}
  />,
  <Page
    number={13}
  />,
  <Page
    number={14}
  />,
  <Page
    number={15}
  />,
  <Page
    number={16}
  />,
  <Page
    number={17}
  />,
  <Page
    number={18}
  />,
  <Page
    number={19}
  />,
  <Page
    number={20}
  />,
]
`;
